reactjs-films-homework/src/components/Header/Form/Form.tsx

```diff
@@ -1,29 +1,86 @@
-import React, { useRef } from 'react';
+import React, { useState } from 'react';
 import search from './assets/search.png';
 import styles from './Form.module.scss';
-const Form: React.FC = () => {
-  const placeholder = 'Movies, person, movie theaters';
-  const ref: any = useRef();
-  const loc = () => {
-    let inputValue = ref.current.value;
-    inputValue ? console.log(inputValue) : console.log('No matches');
-    inputValue = '';
+import { useAppSelector, useAppDispatch } from '../../../hooks/hooks';
+import {
+  setQuery,
+  setActivePage,
+} from '../../../store/rootStore/mainStore/mainSlice';
+import getMainData from '../../../store/rootStore/mainStore/getMaiData/getMainData';
+
+type IFormProps = {
+  placeholder: string;
+};
+
+const Form: React.FC<IFormProps> = ({ placeholder }) => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const pageAfterSearch = '1';
+  const dispatch = useAppDispatch();
+  const [placeholderState, setPlaceholderState] = useState<string>(placeholder);
+  const mainState = {
+    lang: useAppSelector((state) => state.mainReducer.lang),
```

Comment:
It's a good practice to move all the selectors to separate file so that you will be able to reuse them between the components.

Also, when you will have time, please look at reselect lib. It could help you in performance optimization.
https://github.com/reduxjs/reselect